### ISSUE TITLE
Polygon To WKT Not Returning Valid WKT

### ISSUE THREAD

**Issue by jaetoole** [MatanYadaev/laravel-eloquent-spatial#37]
So we seem to have suddenly developed a problem where converting a polygon to WKT gives not valid WKT and it's only recently started happening.
```php
$polygon = new Polygon(
            [
                new LineString([
                    new Point(12.455363273620605, 41.90746728266806),
                    new Point(12.450309991836548, 41.906636872349075),
                    new Point(12.445632219314575, 41.90197359839437),
                    new Point(12.447413206100464, 41.90027269624499),
                    new Point(12.457906007766724, 41.90000118654431),
                    new Point(12.458517551422117, 41.90281205461268),
                    new Point(12.457584142684937, 41.903107507989986),
                    new Point(12.457734346389769, 41.905918239316286),
                    new Point(12.45572805404663, 41.90637337450963),
                    new Point(12.455363273620605, 41.90746728266806),
                ])
            ]
        );
    dd($polygon->toWkt());
```

This returns:
```
POLYGON(LINESTRING(POINT(41.907467282668, 12.455363273621),POINT(41.906636872349, 12.450309991837),POINT(41.901973598394, 12.445632219315),POINT(41.900272696245, 12.4474132061),POINT(41.900001186544, 12.457906007767),POINT(41.902812054613, 12.458517551422),POINT(41.90310750799, 12.457584142685),POINT(41.905918239316, 12.45773434639),POINT(41.90637337
451, 12.455728054047),POINT(41.907467282668, 12.455363273621)))
```

Which is not valid, it should return POLYGON(<coordinates>) which this package has been doing previously. 

Any ideas on how to resolve would be great!

**Comment by MatanYadaev** [comment]
@jaetoole Hi Jae :)
First, the WKT that you just provided hasn't changed since day one, so I'm not sure it's correct that the package changes this format.
Second, can you please share, which WKT you expect? Why the current one isn't good?

**Comment by jaetoole** [comment]
@MatanYadaev Yea I know - that's why I'm confused.

The WKT that I get produced normally is: 
```
POLYGON ((-2.232789 53.484681, -2.231898 53.485211, -2.228755 53.486347, -2.22132 53.489437, -2.217715 53.485211, -2.225761 53.483653, -2.228229 53.482325, -2.232789 53.484681))
```

**Comment by MatanYadaev** [comment]
@jaetoole The WKT format you just mentioned is the more popular one, but both should work. You can see that this package's tests are passing.

I can refactor the package so it will produce this format of WKT, but it will produce breaking changes and will require me to release a major version. 
Before I'm doing so, I want to make sure the issue really exists.

Can you please create a PR that reproduces the issue you just mentioned? I want to play with it.
BTW, which MySQL version do you use?

**Comment by MatanYadaev** [comment]
@jaetoole Fixed in version 2.

**Comment by jaetoole** [comment]
@MatanYadaev can you just check I'm not missing something. I have a factory which creates a record which has a polygon field. When using the factory, it just doesn't work...
```sql
 Integrity constraint violation: 1048 Column 'polygon' cannot be null (SQL: insert into `areas` (`name`, `rps_id`, `polygon`, `updated_at`, `created_at`) values (Oldham, XVA429521, ST_GeomFromText('POLYGON((-2.473898 53.804655,-2.490206 53.808659,-2.512779 53.813778))'), 2022-07-12 08:53:08, 2022-07-12 08:53:08))
```

If I do `select ST_GeomFromText('POLYGON((-2.473898 53.804655,-2.490206 53.808659,-2.512779 53.813778))')` I get null back.

**Comment by jaetoole** [comment]
Also, I tested V2, we're getting errors for Polygon::toWkt() getting 0 parameters etc. It works fine on 1.0.4

**Comment by MatanYadaev** [comment]
@jaetoole 
1. Your polygon isn't valid - the last point in the polygon should be the same as the first one. Try this one: `select ST_GeomFromText('POLYGON((-2.473898 53.804655,-2.490206 53.808659,-2.512779 53.813778,-2.473898 53.804655))');`
2. Which MySQL version do you use? I use MySQL 8 and I don't receive a `null`, I receive this error message: `ERROR: Invalid GIS data provided to function st_geomfromtext.`
3. I fixed `Geometry::toWkt()` in v2.0.1 - now the parameters are optional.
